tripal_chado/src/TripalStorage/ChadoRecords.php

```diff
@@ -687,9 +696,9 @@ public function setLinks(string $base_table) {
 
           // if this column is an ID field and links to this base table then update the value.
           if (array_key_exists($column_alias, $record['link_columns'])) {
-            $base_table =  $record['link_columns'][$column_alias];
-            $record_id = $record_id;
-            $this->setColumnValue($base_table, $table_alias, $delta, $column_alias, $record_id);
+            $base_table = $record['link_columns'][$column_alias];
+            $this->records[$base_table]['tables'][$table_alias]['items'][$delta]['values'][$column_alias] = $record_id;
```

Comment:
It might be good to add a comment here why you didn't use setColumnValue so it doesn't get changed back in the future 🙈